registry/registry_db.go

```diff
@@ -254,132 +199,82 @@ func namespaceSupSubChecks(prefix string) (superspaces []string, subspaces []str
 		UNION
 		SELECT prefix FROM topology WHERE (prefix || '/') LIKE (? || '/%')
 		`
-		args := []interface{}{prefix, prefix}
-		topoSuperSubResults, tmpErr := db.Query(topoSuperSubQuery, args...)
-		if tmpErr != nil {
-			err = tmpErr
+		var results []Topology
+		err = db.Raw(topoSuperSubQuery, prefix, prefix).Scan(&results).Error
+		if err != nil {
 			return
 		}
-		defer topoSuperSubResults.Close()
 
-		for topoSuperSubResults.Next() {
-			// if we make it here, there was a match -- it's a trap!
+		if len(results) > 0 {
+			// If we get here, there was a match -- it's a trap!
 			inTopo = true
 			return
 		}
-		topoSuperSubResults.Close()
 	}
 
 	// Check if any registered namespaces already superspace the incoming namespace,
 	// eg if /foo is already registered, this will be true for an incoming /foo/bar because
 	// /foo is logically above /foo/bar (according to my logic, anyway)
 	superspaceQuery := `SELECT prefix FROM namespace WHERE (? || '/') LIKE (prefix || '/%')`
-	superspaceResults, err := db.Query(superspaceQuery, prefix)
+	err = db.Raw(superspaceQuery, prefix).Scan(&superspaces).Error
 	if err != nil {
 		return
 	}
-	defer superspaceResults.Close()
-
-	for superspaceResults.Next() {
-		var foundSuperspace string
-		if err := superspaceResults.Scan(&foundSuperspace); err == nil {
-			superspaces = append(superspaces, foundSuperspace)
-		}
-	}
 
 	// Check if any registered namespaces already subspace the incoming namespace,
 	// eg if /foo/bar is already registered, this will be true for an incoming /foo because
 	// /foo/bar is logically below /foo
 	subspaceQuery := `SELECT prefix FROM namespace WHERE (prefix || '/') LIKE (? || '/%')`
-	subspaceResults, err := db.Query(subspaceQuery, prefix)
+	err = db.Raw(subspaceQuery, prefix).Scan(&subspaces).Error
 	if err != nil {
 		return
 	}
-	defer subspaceResults.Close()
-
-	for subspaceResults.Next() {
-		var foundSubspace string
-		if err := subspaceResults.Scan(&foundSubspace); err == nil {
-			subspaces = append(subspaces, foundSubspace)
-		}
-	}
 
 	return
 }
 
 func namespaceExistsById(id int) (bool, error) {
-	checkQuery := `SELECT id FROM namespace WHERE id = ?`
-	result, err := db.Query(checkQuery, id)
-	if err != nil {
-		return false, err
-	}
-	defer result.Close()
-
-	found := false
-	for result.Next() {
-		found = true
-		break
+	var namespaces []Namespace
+	result := db.Limit(1).Find(&namespaces, id)
+	if result.Error != nil {
+		return false, result.Error
+	} else {
+		return result.RowsAffected > 0, nil
 	}
-	return found, nil
 }
 
 func namespaceExistsByPrefix(prefix string) (bool, error) {
```

Comment:
You found a bug in the code.. I think we had a fairly inconsistent usage of `namespaceExists()` and `namespaceExistsByPrefix()` throughout various registry APIs. We should merge the two at some point, but it will introduce complication in inserting and updating registration (which sounds like a separate PR)